ui/src/app/pages/workspace/research-purpose.tsx

```diff
@@ -41,51 +56,83 @@ export const ResearchPurpose = withCurrentWorkspace()(
       };
     }
 
-    getSelectedPopulationsSlice(left: boolean) {
-      const populations = getSelectedPopulations(this.props.workspace.researchPurpose);
-      const populationsHalfLen = sliceByHalfLength(populations);
-      if (left) {
-        return populations.slice(0, populationsHalfLen);
-      } else {
-        return populations.slice(populationsHalfLen);
-      }
+    // TODO: Move this to an abstract function and also change the admin workspace to use it
+    getSelectedPopulationsOfInterest() {
+      const researchPurpose = this.props.workspace.researchPurpose;
+      const categories = SpecificPopulationItems.filter(specificPopulationItem => specificPopulationItem
```

Comment:
nit: we can move this 
const categories = SpecificPopulationItems.filter(specificPopulationItem => specificPopulationItem
 

Multi-line suggestions are here!

You can now suggest code changes to multiple lines. Learn more.

 
we can move this to the workspace-edit-text.tsx

      const categories = SpecificPopulationItems.filter(specificPopulationItem => specificPopulationItem
  .subCategory.filter(item => researchPurpose.populationDetails.includes(item.shortName)).length > 0);
      categories.forEach(category => category.subCategory = category.subCategory
        .filter(subCategoryItem => researchPurpose.populationDetails.includes(subCategoryItem.shortName)));
to the workspace-edit-text.tsx ?

Reply:
Talked offline. This ideally would go in `app/utils/research-purpose.tsx` but I want to be able to verify that it works in workspace admin first. (That's what the todo is for.)